dahyun1226/app/src/main/java/com/example/toyproject2020mvvm/ui/recyclerview/ItemAdapter.kt

```diff
@@ -6,50 +6,40 @@ import android.view.LayoutInflater
 import android.view.View
 
 import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
 
 import androidx.recyclerview.widget.RecyclerView
 import com.bumptech.glide.Glide
 import com.example.toyproject2020mvvm.R
+import com.example.toyproject2020mvvm.databinding.RecyclerviewItemBinding
 import com.example.toyproject2020mvvm.ui.RepositoryDetailActivity.Companion.EXTRA_FULL_NAME
 import com.example.toyproject2020mvvm.model.data.GithubRepoData
 import com.example.toyproject2020mvvm.ui.RepositoryDetailActivity
+import com.example.toyproject2020mvvm.viewmodel.MainViewModel
 import kotlinx.android.synthetic.main.recyclerview_item.view.*
 
 
-class ItemAdapter(private var items: ArrayList<GithubRepoData>) :
+class ItemAdapter(private val viewModel: MainViewModel) :
     RecyclerView.Adapter<ItemAdapter.ViewHolder>() {
 
-    override fun getItemCount() = items.size
+    override fun getItemCount() = viewModel.repoData.size
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        val item = items[position]
-        val listener = View.OnClickListener {
-            val intent = Intent(it.context, RepositoryDetailActivity::class.java)
-            intent.putExtra(EXTRA_FULL_NAME, item.fullName)
-            it.context.startActivity(intent)
-        }
-        holder.bind(listener, item)
+        holder.bind(viewModel, position);
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-        val inflatedView =
-            LayoutInflater.from(parent.context).inflate(R.layout.recyclerview_item, parent, false)
-        return ViewHolder(inflatedView)
+        val binding =
+            RecyclerviewItemBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return ViewHolder(binding)
     }
 
-    class ViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {
-        fun bind(listener: View.OnClickListener, item: GithubRepoData) {
-            view.tv_recyclerview_item_FullName.text = item.fullName
-            if (item.language == null) {
-                view.tv_recyclerview_item_Language.text =
-                    view.context.getString(R.string.no_language)
-            } else {
-                view.tv_recyclerview_item_Language.text = item.language
-            }
-            view.setOnClickListener(listener)
-            Glide.with(view).load(item.owner.avatarUrl)
-                .into(view.iv_recyclerview_item_GitAvatarImage)
+    class ViewHolder(private val binding: RecyclerviewItemBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(viewModel: MainViewModel, pos: Int) {
+            binding.viewModel = viewModel
+            binding.pos = pos;
```

Comment:
pos를 넘기는 것보다는 아이템을 넘기는게 더 좋아보입니다.

```suggestion
            binding.repo = viewModel.repoData[pos]
```

여기서 viewModel을 바인딩 해주는데 단순 `seeDetailRepo`함수 사용을 위해 사용하고 있습니다. 좀더 viewModel을 역활과 책임에 맞게 세분화 해서 작업해주세요.

Reply:
2f814fc208d3cf5d57c8596612ffc802fef82c2b
뷰모델 그 자체를 바인딩 하지 않도록 바꿨습니다

Reply:
추가적으로 적용하면 좋은 부분 코멘트 남겼습니다.

Reply:
404f71f
확인했습니다!